From e83edfec1c6fd5b6764a84dfa3eb9d76205b1018 Mon Sep 17 00:00:00 2001
From: Yi Qingliang <niqingliang2003@gmail.com>
Date: Fri, 15 Apr 2016 11:25:39 +0000
Subject: [PATCH] re

Signed-off-by: Yi Qingliang <niqingliang2003@gmail.com>
---
 Makefile | 11 ++++-------
 1 file changed, 4 insertions(+), 7 deletions(-)

diff --git a/Makefile b/Makefile
index cf0c364..c5ac920 100644
--- a/Makefile
+++ b/Makefile
@@ -159,13 +159,10 @@ install-common-files: $(OSC) $(PLUGINS)
 	install ./block_diagrams/* $(PLIB)/block_diagrams
 
 install-all: install-common-files
-	xdg-icon-resource install --noupdate --size 16 ./icons/osc16.png adi-osc
-	xdg-icon-resource install --noupdate --size 32 ./icons/osc32.png adi-osc
-	xdg-icon-resource install --noupdate --size 64 ./icons/osc64.png adi-osc
-	xdg-icon-resource install --noupdate --size 128 ./icons/osc128.png adi-osc
-	xdg-icon-resource install --size 256 ./icons/osc256.png adi-osc
-	xdg-desktop-menu install adi-osc.desktop
-	ldconfig
+	for _size in 16 32 64 128 256; do \
+		install -Dm644 ./icons/osc$${_size}.png $(DESTDIR)/usr/share/icons/hicolor/$${_size}x$${_size}/apps/adi-osc; \
+	done
+	install -Dm644 adi-osc.desktop ${DESTDIR}/usr/share/applications/adi-osc.desktop
 
 uninstall-common-files:
 	rm -rf $(PLIB) $(PSHARE) $(DESTDIR)$(PREFIX)/bin/$(OSC) $(DESTDIR)$(PREFIX)/lib/$(LIBOSC)
-- 
2.7.3
